### DISCUSSION TITLE
Spaceship get_beta_feedback limits don't work

### DISCUSSION THREAD

**Discussion by JasperJhons** [fastlane/fastlane#17128] [Q&A]
Hi everyone!

I try to fetch beta feedbacks via Spaceship's `get_beta_feedback` function, but `limit` is ignored and I got all data at once.
It doesn't work or I do something wrong? (see code below)

```
lane :connect_feedback do
  fastlane_require 'spaceship'
  require 'json'

  Spaceship::Tunes.login
  Spaceship::Tunes.select_team

  # Gets app
  app = Spaceship::ConnectAPI::App.find(ENV["TEST_APP_BUNDLE"])

  # Gets feedback for an app (default includes screenshots and tester info)

  feedbacks = app.get_beta_feedback(limit:61, sort:"-timestamp")

  results = []

  # Iterate over feedbacks
  feedbacks.each do |feedback|
    results.push(feedback.to_json)
  end

  File.open("test.txt", "w+") do |f|
    results.each { |element| f.puts(element) }
  end
end
```

**Comment by abardallis** [comment]
@JasperJhons 

I think I found out what's going on here. 

### Explanation
In `spaceship/lib/spaceship/connect_api/models/app.rb` line 326 we see this: 

```
def get_beta_feedback(client: nil, filter: {}, includes: "tester,build,screenshots", limit: nil, sort: nil)
        client ||= Spaceship::ConnectAPI
        filter ||= {}
        filter["build.app"] = id

        resps = client.get_beta_feedback(filter: filter, includes: includes, limit: limit, sort: sort).all_pages
        return resps.flat_map(&:to_models)
end
```

In that second to last line, `.all_pages` is called (defined on line 52 of `spaceship/lib/spaceship/connect_api/response.rb`). Follow the logic through that file and you'll see that if there is a `"next"` url in the `"links"` object of the JSON response to the first call, it keeps making calls until it runs out of "next" urls to call. 

This ignores the fact that you said `limit:1`. 

### Solution
The solution would be to just bypass this method and call it yourself. Couple notes though: 
* you need to add a `filter:` for `build.app` and pass in your own app id. Because you already set `app = Spaceship::ConnectAPI::App.find(ENV["TEST_APP_BUNDLE"])` on the previous line you can just pass it in like this:
`filter: {'build.app' => app.id}`.
* you don't get the default `includes` anymore but you can pass those in yourself. 
* add `.body` on the end to gets the body out of the request and ignore all the other junk

### TL;DR:
Change the line in your code where you make the call to get `feedbacks` to the following and you should be good to go:
```
  feedbacks = Spaceship::ConnectAPI.get_beta_feedback(filter: {'build.app' => app.id}, includes: "tester,build,screenshots", limit: 61, sort: "-timestamp").body
```
